bc_obps/service/data_access_service/user_operator_service.py

```diff
@@ -0,0 +1,62 @@
+from uuid import UUID
+from service.data_access_service.user_service import UserDataAccessService
+from service.data_access_service.operator_service import OperatorDataAccessService
+from registration.models import Operator, User, UserOperator
+from django.db import transaction
+
+
+from registration.models import (
+    Operator,
+    UserOperator,
+)
+
+
+class UserOperatorDataAccessService:
+    def get_user_operator_by_id(user_operator_id: UUID):
+        # brianna this is the optimized one?
+        return UserOperator.objects.select_related('operator').get(id=user_operator_id)
+        # return UserOperator.objects.get(id=user_operator_id)
+
+    def get_approved_admin_users(operator_id: UUID):
+        operator = Operator.objects.get(id=operator_id)
+        approved_user_operators = UserOperator.objects.filter(
+            operator=operator, role=UserOperator.Roles.ADMIN, status=UserOperator.Statuses.APPROVED
+        )
+        approved_users = User.objects.filter(user_guid__in=approved_user_operators.values('user_id'))
+
+        return approved_users
+
+    def get_pending_admin_users(operator_id: UUID):
+        operator = Operator.objects.get(id=operator_id)
+
+        pending_user_operators = UserOperator.objects.filter(
+            operator=operator, role=UserOperator.Roles.ADMIN, status=UserOperator.Statuses.PENDING
+        )
+        pending_users = User.objects.filter(user_guid__in=pending_user_operators.values('user_id'))
+
+        return pending_users
+
+    def get_an_operators_user_operators_by_user_guid(user_guid: UUID):
+        user = UserDataAccessService.get_user_by_guid(user_guid)
```

Comment:
To properly separate things we just pass the user_guid to the service layer and then if we need the full user, we look it using the guid. This feels wasteful because we already have the full user object from the middleware and therefore in the endpoint. So what happens is in the endpoint we pull the guid off the user, pass that into a service, and then in the service we look up the user object again

Reply:
I've passed the `user_guid` instead of the `user` instance for maximum separation of concerns (I'm avoiding doing anything even remotely db-related in the endpoints). I agree that this is an extra lookup though, so I'm going to make a tech debt ticket to explore alternatives (and block it on the endpoint optimization--I don't know how much effect the extra lookup is having at the moment).

Reply:
I think the confusion here is that we don't need the whole `User` object. The only missing piece is the business_guid that we could fetch individually? 

Either that, or we can ask for an entire `User` object as a parameter (since that is an object the Data Access Layer controls, this is not necessarily a bad thing. It just puts the burden on the caller to figure out how to get one and we conveniently have it in a middleware)

Reply:
Yes, most of the time, we don't need the entire `User` object. However, we need to be cautious about when to pass the user object and when to pass only the `user_guid`. This is because adding an extra query to fetch the user could cause performance issues, given the number of functions that rely on the user and their roles. This was the main reason we cached the user instance in the middleware to avoid additional database calls.